FirebaseAuth/Sources/AuthProvider/OAuth/FIROAuthProvider.m

```diff
@@ -358,16 +367,54 @@ - (void)getHeadFulLiteURLWithEventID:(NSString *)eventID
                                            [NSString stringWithFormat:kHeadfulLiteURLStringFormat,
                                                                       authDomain, argumentsString];
                                      }
-                                     if (completion) {
-                                       NSCharacterSet *set =
-                                           [NSCharacterSet URLFragmentAllowedCharacterSet];
-                                       completion(
-                                           [NSURL
-                                               URLWithString:
-                                                   [URLString
-                                                       stringByAddingPercentEncodingWithAllowedCharacters:
-                                                           set]],
-                                           nil);
+                                     if (appCheck) {
+                                       [appCheck
+                                           getTokenForcingRefresh:false
+                                                       completion:^(
+                                                           id<FIRAppCheckTokenResultInterop> _Nonnull tokenResult) {
+                                                         if (tokenResult.error) {
+                                                           FIRLogWarning(
+                                                               kFIRLoggerAuth, @"I-AUT000018",
+                                                               @"Error getting App Check token; "
+                                                               @"using placeholder token "
+                                                               @"instead. Error: %@",
+                                                               tokenResult.error);
+                                                         }
+                                                         NSString *appCheckTokenFragments =
+                                                             [kAppCheckURLFragmentKey
```

Comment:
nit: Since all of the query param keys are inlined here, I think we can do the same and remove the the `kAppCheckURLFragmentKey` constant, and instead just have `[@"#fac=" stringByAppendingString:tokenResult.token];` here.

Reply:
done, thx